Python basics/Lists/Jay_Q3.py

```diff
@@ -0,0 +1,38 @@
+class refills:
+    def __init__(self, d, m, n):
+        self.i = 0
+        self.distance = d
+        self.maxdist = m
+        self.numofstop = n
+        self.refill = 0
+        self.sum = 0
+        self.stops = []
+
+    def inputs(self):
+        for x in range(0, self.numofstop):
+            d = int(input())
+            self.stops.append(d)
+
+    def findrefill(self):
+
+        for self.i in (0, self.numofstop):
+            if ((self.maxdist - self.stops[self.i])<(self.stops[self.i+1]-self.stops[self.i])):

```

Comment:
Here your loop is running beyond loop length 
EX:
if your length is 5 
at the last iteration i=4
 if ((self.maxdist - self.stops[self.i])<(self.stops[self.i+1]-self.stops[self.i])):
here i+1 is used which will run out of index

Reply:
Now I have understood sir.